table/server/master/src/main/java/alluxio/master/table/DbConfig.java

```diff
@@ -55,153 +97,537 @@ public DbConfig(@JsonProperty("bypass") @Nullable BypassEntry bypassEntry) {
+    public BypassTablesSpec deserialize(JsonParser jp, DeserializationContext cxt)
+        throws IOException, JsonProcessingException {
+      TablesEntryDeserializer<TableEntry> deserializer = new TablesEntryDeserializer<>();
+      IncludeExcludeList<TableEntry> list =
+          deserializer.deserializeToList(TableEntry.class, jp, cxt);
+      return new BypassTablesSpec(list);
+    }
+  }
+
+  /**
+   * On top of a regular NamePatternEntry, contains additional partition specification.
+   */
+  @JsonDeserialize(using = TableEntryDeserializer.class)
+  public static class TableEntry extends NamePatternEntry {
+    private final IncludeExcludeList<NamePatternEntry> mPartitions;
 
     /**
+     * Creates an instance with a specific table name and no partition specification.
+     *
      * @param tableName table name
-     * @param partitions partition names
      */
-    @JsonCreator
-    public BypassTableEntry(@JsonProperty("table") String tableName,
-                            @JsonProperty("partitions") Set<String> partitions) {
-      Preconditions.checkArgument(!tableName.isEmpty(), "empty table name");
-      mTableName = tableName;
+    public TableEntry(String tableName) {
+      this(tableName, IncludeExcludeList.empty());
+    }
+
+    /**
+     * Creates an instance with a specific table name and possibly partitions specifications.
+     *
+     * @param tableName table name
+     * @param partitions partitions
+     */
+    public TableEntry(String tableName, IncludeExcludeList<NamePatternEntry> partitions) {
+      super(tableName);
       mPartitions = partitions;
     }
 
     /**
-     * @return table name
+     * Creates an instance from a {@link NamePatternEntry} with no partition specification.
+     * @param namePatternEntry name entry
      */
-    public String getTable() {
-      return mTableName;
+    public TableEntry(NamePatternEntry namePatternEntry) {
+      super(namePatternEntry);
+      mPartitions = IncludeExcludeList.empty();
     }
 
     /**
-     * @return partition names
+     * Returns partition specifications.
+     * @return partitions
      */
-    public Set<String> getPartitions() {
+    public IncludeExcludeList<NamePatternEntry> getPartitions() {
       return mPartitions;
     }
 
+    /**
+     * Returns table name if the entry is not a regex entry.
+     * @return table name, null if the entry is a regex entry
+     */
+    @Nullable
+    public String getTable() {
+      return getName();
+    }
+
     @Override
     public boolean equals(Object other) {
       if (this == other) {
         return true;
-      } else if (other == null) {
+      }
+      if (other == null) {
         return false;
-      } else if (getClass() != other.getClass()) {
+      }
+      if (getClass() != other.getClass()) {
         return false;
       }
-      BypassTableEntry entry = (BypassTableEntry) other;
-      return Objects.equals(mTableName, entry.mTableName);
+      return super.equals(other);
```

Comment:
We do not compare partitions here, because one table cannot have two different partition specifications. When used in `Set`s, two table-partition spec objects with the same table name will be considered the same and either one is preserved.